canonical/vanilla-framework
# Add CSS theme support for Side Navigation

## Done

* Added new CSS theme support for Side Navigation component
* Note:
    * ~Added dark stylings for `Toggle side navigation` button - previously only had light theme in the examples~
    * Added workaround for unsupported dark themed icons, currently only done for chevron for Expandables and Accordion but there are more icons used as seen in `/docs/examples/patterns/side-navigation/icons`

Fixes [WD-7599](https://warthogs.atlassian.net/jira/software/c/projects/WD/boards/970?selectedIssue=WD-7599)

## QA

- Open [demo](https://vanilla-framework-4972.demos.haus/)
- Go to `/docs/examples` and check out `Side navigation /  XXX` examples
- Side navigation theming should be consistent with the current examples in [vanillaframework.io](vanillaframework.io)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots

[if relevant, include a screenshot or screen capture]


[WD-7599]: https://warthogs.atlassian.net/browse/WD-7599?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
